crates/sema/src/ast_passes.rs

```diff
@@ -283,6 +315,10 @@ impl<'ast> Visit<'ast> for AstValidator<'_, 'ast> {
 
     // Intentionally override unused default implementations to reduce bloat.
     fn visit_expr(&mut self, _expr: &'ast ast::Expr<'ast>) -> ControlFlow<Self::BreakValue> {
+        let ast::Expr { kind, .. } = _expr;
+        if let ast::ExprKind::Lit(lit, _) = kind {
+            self.check_underscores_in_number_literals(lit);
+        }
         ControlFlow::Continue(())
     }
 
```

Comment:
Now this also has to recurse into other expressions and types, so remove the comment, the `visit_ty` implementation, and add `walk_expr` at the end

Reply:
✅